include/dsn/c/api_layer1.h

```diff
@@ -785,6 +785,9 @@ extern DSN_API void          dsn_msg_write_next(
 /*! commit the write buffer after the message content is written with the real written size */
 extern DSN_API void          dsn_msg_write_commit(dsn_message_t msg, size_t size);
 
+/*! append a buffer to the message, the "data" can be reinterpreted cast to (blob*)  */
+extern DSN_API void          dsn_msg_write_append(dsn_message_t msg, dsn_blob_t data);
```

Comment:
This is what is ugly about C interface, unfortunately. That is why in certain cases I would prefere ref_counter or instrusive ptr instead of shared-ptr:-(  I'll think about this though to see how we can enable this more elegantly.